File: Python/Course.py
from Mahasiswa import Mahasiswa
from Dosen import Dosen

# Class Course untuk data mata kuliah
class Course:
    
    # Inisialisasi attribut private
    __nama_matkul = ""
    __dosen = Dosen()       # Composite class dosen sebagai objek pada attribut Dosen
    __mahasiswas = []       # List utnuk menyimpan objek Mahasiswa dari kelas Mahasiswa
    
    
    # Constructor class Course
    def __init__(self, nama_matkul = "", dosen = Dosen() , mahasiswa = Mahasiswa()):
        self.__nama_matkul = nama_matkul
        self.__dosen = dosen
        self.__mahasiswas = []                  # Pemngosongan list agar mahasiswa dari proses sebelumnya terhapus sebelum mengconstruct class Course
        self.__mahasiswas.append(mahasiswa)     # Memasukan object mahasiswa sebagai objek pertama pada list
    
    # Setter dan Getter untuk attribut private pada class Course
    def set_nama_matkul(self, nama_matkul):
        self.__nama_matkul = nama_matkul
    
    def set_dosen(self, dosen):
        self.__dosen = dosen
    
    def get_nama_matkul(self):
        return self.__nama_matkul
    
    def get_dosen(self):
        return self.__dosen
    
    # Penambahan fungsi add utnuk menambahkan objek mahasiswa kedalam list sebagai pengganti setter
    def add_mahasiswa(self, mahasiswa):
        self.__mahasiswas.append(mahasiswa)
    
    # Fungsi Getter untuk mendapatkan list mahasiswas
    def get_mahasiswas(self):
        return self.__mahasiswas
    
        

File: C++/Course.cpp
//Include untuk penggunaan string, list, class Mahasiswa, dan class Dosen
#include <iostream>
#include <string>
#include <bits/stdc++.h>
#include "Mahasiswa.cpp"
#include "Dosen.cpp"

//Deklarasi std agar tidak perlu menulisnya kembali pada code
using namespace std;

//Membuat class Course
class Course
{
private:

    //Membuat atribut yang dibutuhkan/ditambahkan dari parent
    string nama_matkul;
    Dosen dosen;
    list<Mahasiswa> mahasiswas;

public:
    
    //Fungsi construct kosong
    Course(){
        this->nama_matkul = "";
        this->dosen = Dosen();
        this->mahasiswas.clear();
    }

    //Fungsi construct dengan parameter
    Course(string nama_matkul, Dosen dosen, Mahasiswa mahasiswa){
        this->nama_matkul = nama_matkul;
        this->dosen = dosen;
        this->mahasiswas.clear();
        this->mahasiswas.push_back(mahasiswa);
    }


    //Setter dan getter tiap atribut yang ada di class ini
    void set_nama_matkul(string nama_matkul)
    {
        this->nama_matkul = nama_matkul;
    }

    void set_dosen(Dosen dosen){
        this->dosen = dosen;
    }

    void add_mahasiswa(Mahasiswa mahasiswa){
        this->mahasiswas.push_back(mahasiswa);
    }

    string get_nama_matkul(){
        return this->nama_matkul;
    }

    Dosen get_dosen(){
        return this->dosen;
    }

    list<Mahasiswa> get_mahasiswas(){
        return this->mahasiswas;
    }

    ~Course(){

    }
};